### ISSUE TITLE
Sudoku check

### ISSUE THREAD

**Issue by informer2016** [prochazkaml/MichalOS#10]
Hi @prochazkaml :smile: Seems you have a new place now? :wink: Nice to see your wonderful project going strong, and many great updates! I tried a fresh image of MichalOS _(again as a part of a BIOS)_, and here are a few suggestions:
**1)** If to go to a "change the wallpaper image" in settings, there is a minor graphical glitch
**2)** Sudoku - please add a feature to verify the correctness of a solved sudoku
**3)** In addition to existing "Boxes" and "Dots" demos, it would be nice if there'd be a "Dead pixels tester" app - where the whole screen could be covered in a single color, and you could switch between the various single colors by the key presses. I think I saw an app like that on `https://www.ultimatebootcd.com/` - "CHZ Monitor-Test"

Attaching this freshly built floppy just in case: [michalos_29aug2021.zip](https://github.com/prochazkaml/MichalOS/files/7073033/michalos_29aug2021.zip)

**Comment by prochazkaml** [comment]
Hi, nice to see you over here! 😁

Thank you for your suggestions! While I was writing this, I have already resolved suggestions 1) and 3), but I don't understand 2). You mean that it should check in realtime while you're solving the sudoku? It already has a mechanism for checking if you have solved it correctly.

**Comment by informer2016** [comment]
@prochazkaml Sudoku says "You win!" if you solve it correctly, but if you solve it incorrectly - it should tell that it's incorrect, but right now it doesn't tell anything at all in this case. + also it would be nice if there'd be an optional mode for a real time check, which you could enable or disable for a true sudoku experience ;)

**Comment by prochazkaml** [comment]
I've just added an option in the Sudoku app to show you in real time whether you made a mistake!

I don't think it's necessary to show you a "You screwed up" dialog at the end when there was a mistake made, since I think that it's quite obvious (you have 0 steps left and you didn't win)...

**Comment by merinfor2021** [comment]
@prochazkaml sorry for a late reply _(Github problems)_.
> I don't think it's necessary to show you a "You screwed up" dialog at the end when there was a mistake made, since I think that it's quite obvious (you have 0 steps left and you didn't win)...

Maybe it would be nice to print a "There is a mistake in the sudoku" message - like which we see in a new "interactive" mode - but even in the normal mode too if a puzzle has been filled completely and solved incorrectly. Since a user never successfully solved a Sudoku before, he doesn't not know there should be a "You win!" pop-up window. 

By the way, I tried solving a Sudoku and got a premature "You win!" before solving it completely (30 free spaces) :

![sudoku_false_win](https://user-images.githubusercontent.com/91990436/136098865-f8a26e4a-d2cc-4aac-abb1-726784fdc66a.png)

**Comment by prochazkaml** [comment]
@merinfor2021

> sorry for a late reply

No worries! I see you've changed accounts though...

> Since a user never successfully solved a Sudoku before

Yeah, you're right, I added it in. It's in the latest commit now.

> By the way, I tried solving a Sudoku and got a premature "You win!"

And so is this. The detection for solved sudokus was so stupidly written, now it's (hopefully) fixed. I haven't encountered a false trigger now.

**Comment by merinfor2021** [comment]
@prochazkaml Seems to be resolved ;-)

**Comment by prochazkaml** [comment]
Alright, awesome!

Sincere apologies for responding... erm... 10 months late, I seem to not have received a notification for your message at all, and I stumbled upon it just now while casually browsing this repo - I'm probably going to finally do some more work on this project, it's been too long. Cannot make any promises, but I did some restructuring earlier today, which should let me do actual work (as until then, my local source tree was full of unpushable junk, so I reorganized them into appropriate gitignore directories).

By the way, forget about all of the crap that I've said about rewriting the OS in something like C and abandoning assembly. I've learnt a lot about hand-optimized assembly coding (and making it not crash that often) - not on x86, but I feel that I might be able to transfer my new skills to this architecture as well. We'll see, but seeing that I can now recognize crap code that I've written in MichalOS just by looking at it, I have reasonably high hopes.

As I already said, I cannot make any promises, but I'd like to personally dabble a bit more into improving this OS (I'll probably start by backporting features from newer MikeOS releases, as there have been a few now). :)
